From 10602dc505cb4a6fedd738df4bb03e1bda9ad704 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Sun, 26 May 2019 21:34:36 +0200
Subject: [PATCH 262/345] MIPS: OCTEON: Always try to allocate 512 MB of 32-bit
 memory.

Allocation of SWIOTLB area can fail with less than this amount.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/cavium-octeon/setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index 590a79aeb7d3..c60cfe5a9298 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -1114,8 +1114,8 @@ void __init plat_mem_setup(void)
 
 	if (system_limit > max_memory)
 		system_limit = max_memory;
-	/* Try to get 256MB (or more) of 32-bit memory */
-	mem_32_size = system_limit <= (16ull * (1ull << 30)) ? 256 * (1 << 20) : 512 * (1 << 20);
+	/* Try to get 512MB of 32-bit memory */
+	mem_32_size = 512 * (1 << 20);
 
 	cvmx_bootmem_lock();
 	limit_max = 0xffffffffull;
-- 
2.25.1

